Add IDE warnings to code analysis ruleset

stephentoub opened the pull request:
- Add all IDEXXXX warnings to ruleset, but as hidden for now
- Enable IDE0073: file header mismatch

(Missing license headers will now finally be caught by the build, at least for src .cs files... baby steps.)

am11 commented on .editorconfig:
```diff
@@ -158,7 +162,7 @@ dotnet_code_quality.ca1802.api_surface = private, internal
 
 # License header
-file_header_template = Licensed to the .NET Foundation under one or more agreements.\nThe .NET Foundation licenses this file to you under the MIT license.\n
+file_header_template = Licensed to the .NET Foundation under one or more agreements.\nThe .NET Foundation licenses this file to you under the MIT license.
```
Even though the trailing `\n` was removed, IDE0073 still expected an empty line after the header in the files which were missing it (which is good for repo-wide consistency). But generally, is this limitation by design that user has no way to ignore empty line after the header?

stephentoub replied:
> Even though the trailing \n was removed, IDE0073 still expected an empty line after the header in the files which were missing it (which is good for repo-wide consistency).

With the `\n`, it actually required an empty comment line, e.g.
```C#
// Licensed to the .NET Foundation under one or more agreements.
// The .NET Foundation licenses this file to you under the MIT license.
// 
```

am11 replied:
Ah ok. It considers the contents of comment without `//`, that's why we need to separate the header comment from the non-header comment. Neat. 👌

stephentoub committed:
Add IDE warnings to code analysis ruleset

- Add all IDEXXXX warnings to ruleset, but as hidden for now
- Enable IDE0073: file header mismatch
